home-excercises-05-2022/src/main/java/io/codelex/streams/practice/StreamsExercise.java

```diff
@@ -0,0 +1,193 @@
+package io.codelex.streams.practice;
+
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * TODO - search for StreamsExerciseTest and make all tests pass
+ */
+public class StreamsExercise {
+
+    static final Random generator = new Random();
+
+    public static List<Integer> returnSquareRoot(List<Integer> numbers) {
+        return numbers.stream()
+                .map(num -> (int) Math.sqrt(num))
+                .toList();
+    }
+
+    public static List<Integer> getAgeFromUsers(List<User> user) {
+        return user.stream()
+                .map(User::getAge)
+                .toList();
+    }
+
+    public static List<Integer> getDistinctAges(List<User> users) {
+        return users.stream()
+                .map(User::getAge)
+                .distinct()
+                .toList();
+    }
+
+    public static List<User> getLimitedUserList(List<User> users, int limit) {
+        return users.stream()
+                .filter(user -> users.indexOf(user) < limit)
+                .toList();
+
+    }
+
+    public static Integer countUsersOlderThen25(List<User> users) {
+        long usersOlder25 = users.stream()
+                .filter(user -> user.getAge() > 25)
+                .count();
+        return Math.toIntExact(usersOlder25);
```

Comment:
Šis labs risinājums, lai konvertētu 👍

Reply:
Šis tad ir ok? :)

Reply:
Jā, tas lielākais labums ir tajā, ka tā metode met kļūdu, ja tas long būs par lielu priekš int.